### PR TITLE
drivers/kw2xrf: Use TMR3 instead TMR4 to detect ACK timeout 

### PR DESCRIPTION
**Pull request by Ilmu011**
_Originally posted by @PeterKietzmann in #12083_
> @Ilmu011 for the TMR3 vs TMR4 discussion, please keep that one separate in an own issue or even better in a fixing PR :-)

This driver currently uses TMR4 to detect ACK timeouts in TR operations. There is a dedicated timer for detecting timeouts in RX-operations and that is TMR3. I am working on #10364 right now and the kw41z driver uses TMR3 for this purpose, which seems to make more sense.

See [section 7.6.2](https://www.nxp.com/docs/en/reference-manual/MKW2xDxxxRM.pdf#page=171) of the Reference Manual "Using T3CMP to Abort an RX operation"

### CODE REVIEW

File: drivers/kw2xrf/kw2xrf_netdev.c [hunk 1]
```diff
@@ -694,21 +695,29 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
...
-        dev->pending_tx--;
-        netdev->event_callback(netdev, NETDEV_EVENT_TX_NOACK);
-        kw2xrf_seq_timeout_off(dev);
-        kw2xrf_set_sequence(dev, dev->idle_state);
+        if (dregs[MKW2XDM_IRQSTS3] & MKW2XDM_IRQSTS3_TMR3IRQ) {
+            /* if the sequence was aborted by timer 3, ACK timed out */
+            DEBUG("[kw2xrf] TC3TMOUT, SEQIRQ, TX failed\n");
+            irqsts1 |= MKW2XDM_IRQSTS1_SEQIRQ;
+            assert(dev->pending_tx != 0);
+            dev->pending_tx--;
```
Comment (benpicco) [inline drivers/kw2xrf/kw2xrf_netdev.c]: The above three lines are common in both cases, you can put them before the `if`.

File: drivers/kw2xrf/kw2xrf_netdev.c [hunk 2]
```diff
@@ -694,21 +695,29 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
...
+            netdev->event_callback(netdev, NETDEV_EVENT_TX_NOACK);
+            kw2xrf_timer3_seq_abort_off(dev);
+            kw2xrf_abort_rx_ops_disable(dev);
+            kw2xrf_set_sequence(dev, dev->idle_state);
```
Comment (benpicco) [inline drivers/kw2xrf/kw2xrf_netdev.c]: I guess this is not the same as `kw2xrf_set_idle_sequence()`?

File: drivers/kw2xrf/kw2xrf_netdev.c (hunk 1 above)
Comment (Ilmu011) [reply drivers/kw2xrf/kw2xrf_netdev.c]: I'll fix this.

File: drivers/kw2xrf/kw2xrf_netdev.c (hunk 2 above)
Comment (Ilmu011) [reply drivers/kw2xrf/kw2xrf_netdev.c]: This is how it originally was before. According to the comments, `kw2xrf_set_idle_sequence()` is specifically for handling interrupts? There is a lot of code duplication with `kw2xrf_set_sequence` too.

Comment (Ilmu011) [reply drivers/kw2xrf/kw2xrf_netdev.c]: Also, line 705-707 could be de-duplicated too.

**Comment by Ilmu011** [comment]
@benpicco It looks like the difference between `kw2xrf_set_idle_sequence()` and  `kw2xrf_set_sequence` is that `kw2xrf_set_idle_sequence()` checks whether there are still pending transmissions and doesn't abort the sequence if that is the case.
I'm using `kw2xrf_set_idle_sequence()` now.

@PeterKietzmann I also removed the dupilcate code and added some comments.

File: drivers/kw2xrf/kw2xrf_netdev.c [hunk 3]
```diff
@@ -694,21 +697,24 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
...
+        } else {
+            DEBUG("[kw2xrf] SEQIRQ\n");
+            netdev->event_callback(netdev, NETDEV_EVENT_TX_COMPLETE);
+        }
+            /* Disallow TMR3IRQ to cancel RX operation */
+            kw2xrf_timer3_seq_abort_off(dev);
+            /* Disable interrupt for TMR3 and reset TMR3IRQ */
+            kw2xrf_abort_rx_ops_disable(dev);
+            /* Go back to idle state */
+            kw2xrf_set_idle_sequence(dev);
```
Comment (benpicco) [inline drivers/kw2xrf/kw2xrf_netdev.c]: You forgot to bring the indentation back one level 😉

Comment (benpicco) [inline drivers/kw2xrf/kw2xrf_netdev.c]: You forgot to bring the indentation back one level 😉

**Comment by benpicco** [comment]
Looks much better & easier to understand now with the smaller if blocks!

File: drivers/kw2xrf/kw2xrf_netdev.c (hunk 3 above)
Comment (Ilmu011) [reply drivers/kw2xrf/kw2xrf_netdev.c]: Damnit! xD

Comment (benpicco) [reply drivers/kw2xrf/kw2xrf_netdev.c]: Just squash directly.

**Comment by benpicco** [comment]
Feel free to squash.

**Comment by Ilmu011** [comment]
Squashed it :)

**Comment by PeterKietzmann** [comment]
@Ilmu011 I think one commit should be enough here. Please squash all together and name it something alike the first commit.

**Comment by Ilmu011** [comment]
@PeterKietzmann Done.

**Comment by PeterKietzmann** [comment]
Congrats to your first RIOT contribution!

**Pull request merged**